HFStrategy/Strategy/Strategy.py

```diff
@@ -62,76 +74,107 @@ def updateCandleData(self, candle):
       self.marketData[dataKey][-1] = candle
     else:
       self.marketData[dataKey] = [candle]
- 
+  
+  def getLastPrice(self, symbol):
+    mtsPrice = self.lastPrice[symbol]
+    return mtsPrice[0], mtsPrice[1]
+
+  def getPosition(self, symbol):
+    return self.positions.get(symbol)
+
+  def addPosition(self, position):
+    self.positions[position.symbol] = position
+  
+  def removePosition(self, position):
+    self.closedPositions += [position]
+    del self.positions[position.symbol]
+  
+  def onTrade(self, trade):
+    price = trade['price']
+    self.updateIndicatorData('trade', price)
+
+    if self.indicatorsReady():
+      self._onPriceUpdate({
+        'mts': trade['mts'],
+        'price': price,
+        'symbol': trade['symbol'],
+        'candle': trade,
+        'type': 'candle'
+      })
+
   def onCandle(self, candle):
     self.addIndicatorData('candle', candle)
     candle['iv'] = self.indicatorValues()
     self.addCandleData(candle)
 
     if self.indicatorsReady():
-      self.onPriceUpdate({
+      self._onPriceUpdate({
         'mts': candle['mts'],
         'price': candle[self.candlePrice],
         'symbol': candle['symbol'],
         'candle': candle,
         'type': 'candle'
       })
+  
 
-  def onSeedCandle(self, candle):
+  # Starts a thread with the given parameters
+  def _startNewThread(self, func):
```

Comment:
This seems to work perfectly. The only problem is that when the backtest finishes on the main thread there are sometimes some other threads that are still handling an order. To solve this I just made it run on the main thread in backtest mode

Reply:
Makes sense, that works :+1: